cookbook/core/type_system/pytorch_types.py

```diff
@@ -0,0 +1,192 @@
+"""
+.. _pytorch_types:
+
+PyTorch Types
+=============
+
+Flyte promotes the use of strongly-typed data to make it easier to write pipelines that are more robust and easier to test.
+Flyte is primarily used for machine learning besides data engineering. To simplify the communication between Flyte tasks, especially when passing
+around tensors and models, we added support for the PyTorch types.
+"""
+
+# %%
+# Tensors & Modules
+# -----------------
+#
+# Many a times, we need to pass around tensors and modules (aka models). In the absence of native type support for PyTorch tensors and modules,
+# Flytekit resorts to using pickle to serialize and deserialize the entities; in fact, pickle is used for any unknown type.
+# This is not very efficient, and hence, we added PyTorch's serialization and deserialization support to the Flyte type system.
+import torch
+
+from flytekit import task, workflow
+
+
+@task
+def generate_tensor_2d() -> torch.Tensor:
+    return torch.tensor([[1.0, -1.0, 2], [1.0, -1.0, 9], [0, 7.0, 3]])
+
+
+@task
+def reshape_tensor(tensor: torch.Tensor) -> torch.Tensor:
+    # convert 2D to 3D
+    tensor.unsqueeze_(-1)
+    return tensor.expand(3, 3, 2)
+
+
+@task
+def generate_module() -> torch.nn.Module:
+    bn = torch.nn.BatchNorm1d(3, track_running_stats=True)
+    return bn
+
+
+@task
+def get_model_weight(model: torch.nn.Module) -> torch.Tensor:
+    return model.weight
+
+
+class MyModel(torch.nn.Module):
+    def __init__(self):
+        super(MyModel, self).__init__()
+        self.l0 = torch.nn.Linear(4, 2)
+        self.l1 = torch.nn.Linear(2, 1)
+
+    def forward(self, input):
+        out0 = self.l0(input)
+        out0_relu = torch.nn.functional.relu(out0)
+        return self.l1(out0_relu)
+
+
+@task
+def get_l1() -> torch.nn.Module:
+    model = MyModel()
+    return model.l1
+
+
+@workflow
+def pytorch_native_wf():
+    reshape_tensor(tensor=generate_tensor_2d())
+    get_model_weight(model=generate_module())
+    get_l1()
+
+
+# %%
+# Passing around tensors and modules is no more a hassle!
+
+# %%
+# Checkpoint
+# ----------
+#
+# ``PyTorchCheckpoint`` is a special type of checkpoint that is used to serialize and deserialize PyTorch models.
```

Comment:
Can we rephrase to "``PyTorchCheckpoint`` is a special type of checkpoint to serialize and deserialize PyTorch models. "?